docs/api/apiv3/endpoints/work-packages.apib

```diff
@@ -126,6 +127,10 @@ and [update](#work-packages-work-package-patch). The attachments the work packag
                         "href": "/api/v3/work_packages/1528/attachments",
                         "method": "post"
                     },
+                    "additionalTabs": {
+                        "href": "/work_packages/1528/additionalTabs/github",
+                        "title": "github"
+                    },
```

Comment:
We may want a better name than `additionalTabs` since "tabs" is not a concept on the API level (but a view thing). I really only need the `title` and added `href` just because links need it. If there is a better way to just provide a list of identifiers, we may want to use that instead.

Reply:
If the additionalTab/integration is not a resource in itself, we could for example make this a part of the work package schema. This behaves very similarly to how the attribute groups (what attributes and in what order are shown on the work package per type) are defined in the backend.

Reply:
If I where to choose freely, I'd simply put it in as a simple link within the `_links` section. Highly abbreviated, this would then look like this:

```
{
  "_links": {
     ...
     "github": {
        "href": "someurl",
        "title": GitHub
     },
     "anotherLink": {
     ...
     },
     ...
   },
  ...
}
```
So no nesting but each added link by itself. 

This would have the benefit of being completely independent from the front end and while also not limiting the type of link from a semantic point of view (if it is not an integration). 

I wouldn't want to place it in the schema as it then always requires fetching the schema and, for reasons beyond the scope of the GitHub-Integration, its presence might depend on the individual work package. To construct an example in the GitHub scope, maybe we later only want it to show up if the work package is in a status >= "in development".

This of course would require the frontend to look through the complete list of links but I don't think that would be a problem.

I also think that rather sooner or later, at least for the GitHub case, the link will effectively reference a resource which e.g. displays or references the commits/PRs.

Reply:
✅ done - a plugin can extend the work package representer to add a link (like my for of the github integration does now), but check on it's own when it wants to be displayed (e.g. by looking at the link)